src/map/zone_entities.cpp

```diff
@@ -350,23 +350,23 @@ void CZoneEntities::FindPartyForMob(CBaseEntity* PEntity)
 
     CMobEntity* PMob = static_cast<CMobEntity*>(PEntity);
 
-    // force all mobs in a burning circle to link
-    ZONE_TYPE zonetype  = m_zone->GetTypeMask();
-    bool      forceLink = zonetype & ZONE_TYPE::DYNAMIS || PMob->getMobMod(MOBMOD_SUPERLINK);
-
-    if ((forceLink || PMob->m_Link || PMob->m_Type & MOBTYPE_BATTLEFIELD) && PMob->PParty == nullptr)
+    bool forceLink = PMob->ShouldForceLink();
+    // check for sublinks even if a family doesn't link with itself
+    int16 sublink = PMob->getMobMod(MOBMOD_SUBLINK);
+    if ((forceLink || PMob->m_Link || sublink) && PMob->PParty == nullptr)
     {
         FOR_EACH_PAIR_CAST_SECOND(CMobEntity*, PCurrentMob, m_mobList)
         {
-            if (!forceLink && !PCurrentMob->m_Link)
+            if (!forceLink && !sublink && !PCurrentMob->m_Link)
             {
                 continue;
             }
 
-            int16 sublink = PMob->getMobMod(MOBMOD_SUBLINK);
-
-            if (PCurrentMob->PParty && PCurrentMob->allegiance == PMob->allegiance &&
-                (forceLink || PCurrentMob->m_Family == PMob->m_Family || (sublink && sublink == PCurrentMob->getMobMod(MOBMOD_SUBLINK))))
+            if (
+                PCurrentMob->PParty && PCurrentMob->allegiance == PMob->allegiance &&
+                (forceLink ||
```

Comment:
```suggestion
                ((forceLink && PCurrentMob->ShouldForceLink()) ||
```

Noticed a quirk with my superlink changes that was putting `forceLink` mobs in the first party it found. Serendipty that your PR comes along with a helper to easily solve the issue!